mobile/screens/resources/Research.tsx

```diff
@@ -1,19 +1,124 @@
-import React from "react";
-import { View, Text, Button } from "react-native";
+import React, { useEffect, useState } from "react";
+import { View, Text, Button, Linking, StyleSheet, Pressable, Image } from "react-native";
 import { ResourcesStackScreenProps } from "../../types";
+import { doc, getDoc, setDoc } from "firebase/firestore";
+import { db } from "../../config/firebase";
+import ConstructionSVG from "../../assets/images/construction";
 
 export default function Research({
   navigation,
 }: ResourcesStackScreenProps<"Research">) {
+  const [description, setDescription] = useState("");
+  const [link, setLink] = useState("");
+  let linkCount = link.length;
+
+  useEffect(() => {
+    let ignore = false;
+    getDoc(doc(db, "resources/research")).then((doc) => {
+      if (!ignore) {
+        setDescription(doc?.data()?.markdown);
+        setLink(doc?.data()?.url);
+      }
+    });
+
+    return () => {
+      ignore = true;
+    };
```

Comment:
Nice use of cleanup functions! Thank you for your hard work getting the app running!